File: solutions/javascript/space-age/1/space-age.js
//
// This is only a SKELETON file for the 'Space Age' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const age = (planet, ageInSeconds) => {
  const earthYearInSeconds = 31557600; // 365.25 days × 24 × 60 × 60
  
  const orbitalPeriods = {
    mercury: 0.2408467,
    venus: 0.61519726,
    earth: 1.0,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
  };
  
  if (!orbitalPeriods.hasOwnProperty(planet)) throw new Error('not a planet');

  const earthYears = ageInSeconds / earthYearInSeconds;
  const planetYears = earthYears / orbitalPeriods[planet];
  
  return parseFloat(planetYears.toFixed(2));
};

File: solutions/fortran/space-age/1/space_age.f90
module space_age
  implicit none
contains

  double precision function age_in_years(planet, seconds)
    character(len=*), intent(in) :: planet
    double precision, intent(in) :: seconds

    double precision :: orbital_period
    double precision, parameter :: seconds_per_earth_year = 31557600.0d0

    select case(trim(planet))
    case("Mercury")
      orbital_period = 0.2408467d0
    case("Venus")
      orbital_period = 0.61519726d0
    case("Earth")
      orbital_period = 1.0d0
    case("Mars")
      orbital_period = 1.8808158d0
    case("Jupiter")
      orbital_period = 11.862615d0
    case("Saturn")
      orbital_period = 29.447498d0
    case("Uranus")
      orbital_period = 84.016846d0
    case("Neptune")
      orbital_period = 164.79132d0
    case default
      ! If an unknown planet is passed, return 0.0 (or you could raise an error)
      orbital_period = 1.0d0  ! fallback, though shouldn't happen
    end select

    age_in_years = seconds / (seconds_per_earth_year * orbital_period)

  end function age_in_years

end module space_age